File: app/src/app/java/com/android/architecture/ArchAppContext.java
package com.android.architecture;


import android.app.Activity;
import android.content.Intent;

import com.android.architecture.launcher.AppLauncherActivity;
import com.app.base.AppContext;

import org.jetbrains.annotations.NotNull;

import dagger.hilt.android.HiltAndroidApp;

/**
 * @author Ztiany
 * Email: ztiany3@gmail.com
 * Date : 2018-11-02 16:32
 */
@HiltAndroidApp
public class ArchAppContext extends AppContext {

    @Override
    public void restartApp(@NotNull Activity activity) {
        Intent intent = new Intent(this, AppLauncherActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }

}

File: app/src/main/java/com/ztiany/architecture/ArchAppContext.kt
package com.ztiany.architecture

import android.content.Context
import android.support.multidex.MultiDex
import com.android.base.app.BaseAppContext

/**
 *
 *@author Ztiany
 *      Date : 2018-09-06 17:14
 */
class ArchAppContext : BaseAppContext() {

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }

}